sqlmesh/core/notification_target.py

```diff
@@ -218,21 +253,51 @@ def send(self, notification_status: NotificationStatus, msg: str, **kwargs: t.An
 
 
 class BaseSlackNotificationTarget(BaseNotificationTarget):
-    def send(self, notification_status: NotificationStatus, msg: str, **kwargs: t.Any) -> None:
+    def send(
+        self,
+        notification_status: NotificationStatus,
+        msg: str,
+        audit_error: t.Optional[AuditError] = None,
+        exc: t.Optional[str] = None,
+        **kwargs: t.Any,
+    ) -> None:
+
         status_emoji = {
             NotificationStatus.PROGRESS: slack.SlackAlertIcon.START,
             NotificationStatus.SUCCESS: slack.SlackAlertIcon.SUCCESS,
             NotificationStatus.FAILURE: slack.SlackAlertIcon.FAILURE,
             NotificationStatus.WARNING: slack.SlackAlertIcon.WARNING,
             NotificationStatus.INFO: slack.SlackAlertIcon.INFO,
         }
+
         composed = slack.message().add_primary_blocks(
             slack.header_block(f"{status_emoji[notification_status]} SQLMesh Notification"),
-            slack.context_block(f"*Status:* {notification_status.value}"),
+            slack.context_block(f"*Status:* `{notification_status.value}`"),
+            slack.divider_block(),
+            slack.text_section_block(f"*Message*: {msg}"),
+        )
+
+        details = []
+        if audit_error:
+            details = [
+                slack.fields_section_block(
+                    f"*Audit*: `{audit_error.audit_name}`",
+                    f"*Model*: `{audit_error.model_name}`",
+                    f"*Count*: `{audit_error.count}`",
+                ),
+                slack.preformatted_rich_text_block(str(audit_error.query)),
```

Comment:
Instead of doing
```
str(audit_error.query)
```
You can do:
```
audit_error.sql(dialect=audit_error.audit.dialect)
```
If you also want to format the query you can do:
```
audit_error.sql(dialect=audit_error.audit.dialect, pretty=True)
```

Reply:
Please note that `audit_error.model` is optional, so the `None` check might be required.

Reply:
@izeigerman thanks for the guidance.

`AuditError` doesn't seem to hold on to `Audit` (see [here](https://github.com/TobikoData/sqlmesh/blob/ce8e4c72d90b0cfda22bfbbb6dcceb577af68898/sqlmesh/utils/errors.py#L64-L75)). Should I:

1. add an `audit: Audit` field
1. add a `dialect: str` field and populate it from `audit.dialect`
1. add a `dialect: str` field and populate it from `engine_adapter.dialect`
1. add a `dialect: str` field and populate it from `snapshot.model.dialect`

All of these options would happen around [here](https://github.com/TobikoData/sqlmesh/blob/ce8e4c72d90b0cfda22bfbbb6dcceb577af68898/sqlmesh/core/snapshot/evaluator.py#L768)

Reply:
I suggest the engine_adapter dialect since that would be the dialect used to actually run the audit. Another option would be to stored the rendered audit query in the error, but then you'd lose the flexibility to format it (I guess you could parse it back into a sqlglot expression again).

Reply:
Thanks @vchan 

I've added 2 commits:

* https://github.com/TobikoData/sqlmesh/pull/2144/commits/77461fb79cba4e25378625dec2f380d9a29c6fe7 which captures the engine adapter's dialect and exposes a method to print the sql statement to a string
* https://github.com/TobikoData/sqlmesh/pull/2144/commits/5ef9f6e55f0a62870fbb6a03f516e8ef9a94410c that uses this new feature for the slack message

PTAL